enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,685 @@
+
+As a user, I run a CI platform where I have capacity constraints for my batch jobs. 
+I want to be able to set quota limits across various tenants. 
+I am using core openshift components like a batch-job. 
+I would like to use Kueue and Jobs for basic job quota management.
+
+Generally I want fair sharing, preemption and other common abilities in other job schedulers such as SLURM or LSF.
+
+#### AI Training
+
+As a user, I want the ability to run distributed jobs with some kind of quota management solution. 
+AI users tend to use higher level workloads like Ray, Kubeflow Training Operator or JobSet. 
+Kueue should provide the ability handle these workloads.
+
+#### Scheduling Batch Workloads Across Multiple Clusters
+
+Kueue provides support for dispatching jobs across multiple clusters. I would like to use Kueue as a way to manage quotas for batch jobs with multiple clusters.
+
+OCM provides a POC for how to use [MultiKueue](https://github.com/open-cluster-management-io/ocm/tree/main/solutions/kueue-admission-check).
+
+We are considering MultiKueue out of scope for tech preview.
+
+#### Build Platforms
+
+
+Workflows/Pipelines can also be resource constrained. 
+One area can be the usage of Kueue’s Pod Integration to preemption workflows in Konflux. 
+Openshift operators builds could take precedence over community pipelines. 
+Another area can be limiting the amount of pipelines that can execute so we don’t flood the clusters with pending requests.
+
+The internal Tekton team is looking at an integration with Kueue.
+
+- [Upstream Issue](https://github.com/kubernetes-sigs/kueue/issues/4167)
+- [Jira Epic](https://issues.redhat.com/browse/SECFLOWOTL-242)
+
+#### Autoscaling
+
+Kueue was designed to allow for the use of quota management with autoscaling. 
+Provisioning Request (abbr. ProvReq) is a new namespaced Custom Resource that aims to allow users to ask CA for capacity for groups of pods. 
+Kueue provides integration with ProvReq so that if there is quota constrains on the cluster, one could trigger ProvReq to create capacity at the workload level.
+
+Node Autoscaling is working on ProvReq.
+
+A kueue admin can set up a ClusterQueue so that users that have access to that cluster queue via their localqueue could trigger a ProvReq.
+This ProvReq would have the cluster autoscaler create nodes that can fit the workload.
+
+This functionality is possible once [ProvRequest in Autoscaling](https://issues.redhat.com/browse/OCPSTRAT-1331) is done.
+
+
+### Goals
+
+- Provide a Red Hat Certified Operator for Kueue Deployments
+- Provide an API so that users can configure Kueue with their use case in mind.
+- Kueue will be forked and maintained as [openshift/kubernetes-sig-kueue](https://github.com/openshift/kubernetes-sigs-kueue)
+- [KueueOperator](https://github.com/openshift/kueue-operator) will be created to manage the installation and configuration of Kueue 
+
+### Non-Goals
+
+- UX improvements on the Kueue APIs as part of the operator.
```

Comment:
Though UX is out - we would have to discuss on how we would like to migrate Distributed Workloads UI which majorly has Kueue metrics. We can keep that as a part of RHOAI, but it probably makes sense to expose those metrics in OCP dashboard instead for standalone installations of Kueue without RHOAI.

Reply:
I think you could maybe reuse the same Prometheus Rules that you use for these metrics.

Reply:
I agree but I'm not sure every customer installation would want the same prometheus rules that RHOAI has.

This bullet point was mostly in mind for how to improve Kueue APIs or add easier ways to create kueue resources. We want to call that out of scope for this operator.

Metrics will be exported into OCP dashboard as part of our deployment. What RHOAI chooses to build for prometheus rules or dashboards may need some discussion if its general for all kinds of Kueue installations or tailor made for RHOAI.